### ISSUE TITLE
iter_bucket fails in AWS Lambda function

### ISSUE THREAD

**Issue by amijalis** [RaRe-Technologies/smart_open#605]
I am trying to use the iter_bucket() function in a dockerized Python 3.9 Lambda function.

I see that this issue #340 has been considered in the past, and that there is the following code in smart_open.concurrency to try and prevent this error from occurring.

```
_MULTIPROCESSING = False
try:
    import multiprocessing.pool
    _MULTIPROCESSING = True
except ImportError:
    warnings.warn("multiprocessing could not be imported and won't be used")
```

However, I find that in my case, importing multiprocessing.pool does not raise the ImportError. Therefore, it appears that _MULTIPROCESSING is being set to True and then the following OSError gets raised at runtime:

```
 "errorMessage": "[Errno 38] Function not implemented",
  "errorType": "OSError",
  "stackTrace": [
    "  File \"/home/app/app.py\", line 67, in handler\n    raise ex\n",
    "  File \"/home/app/app.py\", line 60, in handler\n    result = function(**params)\n",
    "  File \"/home/app/app.py\", line 268, in func\n    key,content = next(bam)\n",
    "  File \"/usr/local/lib/python3.9/site-packages/smart_open/s3.py\", line 1172, in iter_bucket\n    with smart_open.concurrency.create_pool(processes=workers) as pool:\n",
    "  File \"/usr/local/lib/python3.9/contextlib.py\", line 117, in __enter__\n    return next(self.gen)\n",
    "  File \"/usr/local/lib/python3.9/site-packages/smart_open/concurrency.py\", line 68, in create_pool\n    pool = multiprocessing.pool.Pool(processes=processes)\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/pool.py\", line 191, in __init__\n    self._setup_queues()\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/pool.py\", line 343, in _setup_queues\n    self._inqueue = self._ctx.SimpleQueue()\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/context.py\", line 113, in SimpleQueue\n    return SimpleQueue(ctx=self.get_context())\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/queues.py\", line 342, in __init__\n    self._rlock = ctx.Lock()\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/context.py\", line 68, in Lock\n    return Lock(ctx=self.get_context())\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/synchronize.py\", line 162, in __init__\n    SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)\n",
    "  File \"/usr/local/lib/python3.9/multiprocessing/synchronize.py\", line 57, in __init__\n    sl = self._semlock = _multiprocessing.SemLock(\n"
  ]
```

Is it possible to manually set the _MULTIPROCESSING  when starting an iter_bucket() operation? Or is there another hack to get this working?

Thanks!

Versions:
Linux-5.4.72-microsoft-standard-WSL2-x86_64-with-glibc2.28
Python 3.9.4 (default, Apr 10 2021, 15:31:19)
[GCC 8.3.0]
smart_open 5.0.0

**Comment by mpenkov** [comment]
> Is it possible to manually set the _MULTIPROCESSING when starting an iter_bucket() operation? Or is there another hack to get this working?

Maybe. Have you tried?

**Comment by akovari** [comment]
Seems like this works:

```python
import smart_open.concurrency
smart_open.concurrency._MULTIPROCESSING = False
```

**Comment by mpenkov** [comment]
Looks like there's an easy workaround, closing.

**Comment by ronkorving** [comment]
Wasn't this supposed to have been solved by #368 ?
I also ran into this today, which is how I found both this issue and the apparent solution using futures.
